libp2p/Session.cpp

```diff
@@ -248,6 +255,24 @@ void Session::write()
             DEV_GUARDED(x_framing)
             {
                 m_writeQueue.pop_front();
+                if(m_writeQueue.empty() && disconnectImminent)
+                {
+                    ba::io_context ioc;
+                    ba::steady_timer disconnectTimer(ioc);
+                    disconnectTimer.expires_from_now(chrono::seconds(2));
+                    disconnectTimer.async_wait([this](boost::system::error_code const& ) {});
+                    ioc.run();
```

Comment:
You don't need to explicitly call `run`, Host already does this (see `Host::doWork`)